packages/safe-tools-client/app/components/schedule-payment-form-action-card/index.gts

```diff
@@ -378,22 +378,41 @@ export default class SchedulePaymentFormActionCard extends Component<Signature>
 
     this.scheduleErrorMessage = undefined;
 
-    try {
-      const { currentSafe } = this.safes;
+    const { currentSafe } = this.safes;
+
+    let payAt = this.selectedPaymentType === 'one-time' ? Math.round(this.paymentDate!.getTime() / 1000) : null;
+    let recurringDayOfMonth = this.selectedPaymentType === 'monthly' ? this.paymentDayOfMonth! : null;
+    let recurringUntil = this.selectedPaymentType === 'monthly' ? Math.round(this.monthlyUntil!.getTime() / 1000) : null
+
+    let params = {
+      safeAddress: currentSafe.address,
+      spModuleAddress: currentSafe.spModuleAddress,
+      tokenAddress: this.paymentTokenQuantity.address,
+      amount: this.paymentTokenQuantity.count,
+      payeeAddress: this.payeeAddress,
+      executionGas: Number(this.gasEstimation.gas),
+      maxGasPrice: maxGasPriceString,
+      gasTokenAddress: this.selectedGasToken.address,
+      salt,
+      payAt,
+      recurringDayOfMonth,
+      recurringUntil
+    }
 
+    try {
       yield taskFor(this.scheduledPaymentSdk.schedulePayment).perform(
-        currentSafe.address,
-        currentSafe.spModuleAddress,
-        this.paymentTokenQuantity.address,
-        this.paymentTokenQuantity.count,
-        this.payeeAddress,
-        Number(this.gasEstimation.gas),
-        maxGasPriceString,
-        this.selectedGasToken.address,
-        salt,
-        this.selectedPaymentType === 'one-time' ? Math.round(this.paymentDate!.getTime() / 1000) : null,
-        this.selectedPaymentType === 'monthly' ? this.paymentDayOfMonth! : null,
-        this.selectedPaymentType === 'monthly' ? Math.round(this.monthlyUntil!.getTime() / 1000) : null,
+        params.safeAddress,
+        params.spModuleAddress,
+        params.tokenAddress,
+        params.amount,
+        params.payeeAddress,
+        params.executionGas,
+        params.maxGasPrice,
+        params.gasTokenAddress,
+        params.salt,
+        params.payAt,
+        params.recurringDayOfMonth,
+        params.recurringUntil,
```

Comment:
supernit:  if you want to avoid mapping the params by hand and they are in the correct order you could do something like: 
`perform(...Object.values(params))`

but I also understand that this might be too magical, so I'm ok if this suggestion is ignored

Reply:
Cool - that's good to know. I'm gonna merge and keep in mind when we revisit it.